app/assets/javascripts/services.js

```diff
@@ -210,12 +213,17 @@ angular.module('HNSearch.services', ['algoliasearch', 'ngStorage', 'angular-goog
         // typo tolerance
         this.params.typoTolerance = settings.sort === 'byPopularity' && settings.typoTolerance;
 
-        // story text
-        if (!settings.storyText) {
+        // restrict attributes to search on, based on storyText and authorText settings
+        this.params.restrictSearchableAttributes = ['title', 'comment_text', 'url'];
+        if (settings.storyText) {
+            this.params.restrictSearchableAttributes.push('story_text');
+        } else{
             $location.search('storyText', false);
-            this.params.restrictSearchableAttributes = ['title', 'comment_text', 'url', 'author'];
+        }
+        if (settings.authorText) {
+            this.params.restrictSearchableAttributes.push('author');
         } else {
-            this.params.restrictSearchableAttributes = [];
+            $location.search('authorText', false);
         }
```

Comment:
For performance reason, I'm gonna change a bit how this logic is done: we rather want to set `[]` instead of the full list of attributes.